addons/stock/models/stock_picking.py

```diff
@@ -799,6 +799,20 @@ def _onchange_picking_type(self):
                     'message': partner.picking_warn_msg
                 }}
 
+    @api.onchange('location_id')
+    def _onchange_location_id(self):
+        if self.env.context.get('do_not_unreserve') is not None:
+            return
```

Comment:
I don't think this can happen, as `onchange` methods are only triggered through the client.